src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java

```diff
@@ -0,0 +1,386 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pesmypetcare.webservice.entity.WeightEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class WeightDaoTest {
+    private static List<Map<String, Object>> weightList;
+    private static WeightEntity weightEntity;
+    private static String date;
+    private static String date2;
+    private static String owner;
+    private static String petName;
+    private static Double value;
+    private static String field;
+    private static final String USERS_KEY = "users";
+    private static final String PETS_KEY = "pets";
+    private static final String WEIGHTS_KEY = "weights";
+    private final String EXCECUTION_EXC_MSG = "Should throw DatabaseAccessException when ExecutionException received";
+    private final String INTERRUPTED_EXC_MSG = "Should throw DatabaseAccessException when InterruptedException "
+        + "received";
+    private final String DOCUMENT_NOT_EXISTS_EXC_MSG = "Should throw DatabaseAccessException when Weight document"
+        + " doesn't exist";
+
+    @Mock
+    private Firestore db;
+    @Mock
+    private CollectionReference usersRef;
+    @Mock
+    private DocumentReference ownerRef;
+    @Mock
+    private CollectionReference petsRef;
+    @Mock
+    private DocumentReference petRef;
+    @Mock
+    private CollectionReference weightsRef;
+    @Mock
+    private DocumentReference weightRef;
+    @Mock
+    private ApiFuture<QuerySnapshot> futureQuery;
+    @Mock
+    private QuerySnapshot querySnapshot;
+    @Mock
+    private ApiFuture<DocumentSnapshot> futureDocument;
+    @Mock
+    private DocumentSnapshot documentSnapshot;
+    @Mock
+    private List<QueryDocumentSnapshot> weightsDocuments;
+    @Mock
+    private Iterator<QueryDocumentSnapshot> it;
+
+    @InjectMocks
+    private WeightDao weightDao = new WeightDaoImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        weightList = new ArrayList<>();
+        weightEntity = new WeightEntity();
+        date = "2019-10-22T00:47:00";
+        date2 = "2021-10-22T00:47:00";
+        owner = "PericoDeLosPalotes";
+        petName = "TupoJohn";
+        field = "weightValue";
+        value = 9.0 / 2.0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['9.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)